src/EditorFeatures/CSharp/EventHookup/EventHookupCommandHandler_TabKeyCommand.cs

```diff
@@ -222,24 +223,30 @@ private void GenerateAndAddEventHandler(ITextView textView, ITextBuffer subjectB
             var root = document.Root;
             var eventHookupExpression = root.GetAnnotatedNodesAndTokens(plusEqualsTokenAnnotation).Single().AsToken().GetAncestor<AssignmentExpressionSyntax>();
 
-            var generatedMethodSymbol = GetMethodSymbol(document, eventHandlerMethodName, eventHookupExpression, cancellationToken);
+            var typeDecl = eventHookupExpression.GetAncestor<TypeDeclarationSyntax>();
+            var methodKind = typeDecl is null
+                ? MethodKind.LocalFunction
+                : MethodKind.Ordinary;
+
+            var generatedMethodSymbol = GetMethodSymbol(document, eventHandlerMethodName, eventHookupExpression, methodKind, cancellationToken);
 
             if (generatedMethodSymbol == null)
             {
                 return null;
             }
 
-            var typeDecl = eventHookupExpression.GetAncestor<TypeDeclarationSyntax>();
+            var container = (SyntaxNode)typeDecl ?? eventHookupExpression.GetAncestor<CompilationUnitSyntax>();
 
-            var typeDeclWithMethodAdded = CodeGenerator.AddMethodDeclaration(typeDecl, generatedMethodSymbol, document.Project.Solution.Workspace, new CodeGenerationOptions(afterThisLocation: eventHookupExpression.GetLocation()));
+            var newContainer = CodeGenerator.AddMethodDeclaration(container, generatedMethodSymbol, document.Project.Solution.Workspace, new CodeGenerationOptions(afterThisLocation: eventHookupExpression.GetLocation()));
 
-            return root.ReplaceNode(typeDecl, typeDeclWithMethodAdded);
+            return root.ReplaceNode(container, newContainer);
         }
 
         private static IMethodSymbol GetMethodSymbol(
             SemanticDocument semanticDocument,
             string eventHandlerMethodName,
             AssignmentExpressionSyntax eventHookupExpression,
+            MethodKind methodKind,
```

Comment:
@davidwengier While this change makes sense, I wonder how things were working in https://github.com/dotnet/roslyn/pull/50726 without this change?

Reply:
hmm.. good question. Perhaps a change in the code generation bits since then?

Apologies for not noticing the existing issue/pr

Reply:
> Perhaps a change in the code generation bits since then?

I took a quick look from sourceroslyn.io and can't really find how this can affect anything. Going to open a quick draft to see what exactly fails as I'm curious.

> Apologies for not noticing the existing issue/pr

No problem!

Reply:
Wouldn't be surprised if it's unnecessary, but I definitely saw a failure at one point where the access modifier was added to the local function, which obviously was wrong. I can't remember exactly what the rest of the code looked like at that point though :)